Kernel/Storage/NVME/NVMEController.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) 2021, Pankaj Raghav <pankydev8@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "NVMEController.h"
+#include "AK/Format.h"
+#include <AK/RefPtr.h>
+#include <AK/Types.h>
+#include <Kernel/Arch/x86/IO.h>
+#include <Kernel/Arch/x86/Processor.h>
+#include <Kernel/Bus/PCI/API.h>
+#include <Kernel/Devices/Device.h>
+#include <Kernel/FileSystem/ProcFS.h>
+#include <Kernel/Sections.h>
+
+namespace Kernel {
+
+NonnullRefPtr<NVMEController> NVMEController::initialize(const Kernel::PCI::DeviceIdentifier& device_identifier)
+{
+    return adopt_ref(*new NVMEController(device_identifier));
```

Comment:
Thanks for the comment! I tried to keep the pattern consistent across storage devices. But I totally agree with your point. I would like to hear @supercomputer7 view as well. 
But I will try to take your comments into account in my next revision.

Reply:
@ADKaster is right here, we should write new kernel code with OOM mindset. It will take time to propagate these patterns but eventually we will be able to handle OOM much better than today :)